packages/edit-site/src/components/header/style.scss

```diff
@@ -39,15 +38,43 @@
 	}
 }
 
+// This element is used to do the same transition as
+// the navigation sidebar. This element is at least
+// `$header-height` wide, when the navigation panel is
+// opened then it widens to the same width as the
+// navigation panel. While doing that, it also
+// moves itself to the left (negative `margin-left`)
+// to avoid content jumping
+.edit-site-header__navigation-panel-spacer {
+	min-width: $header-height;
+	transition: margin-left 100ms linear;
+	margin-left: 0;
+	@include reduce-motion("transition");
+}
+
+.edit-site-header__navigation-panel-spacer::before {
+	content: "";
+	display: block;
+	width: 0;
+	transition: width 100ms linear;
+	@include reduce-motion("transition");
+}
+
```

Comment:
I'm not super into this spacer... 😅 

I've tried animating the header padding instead, and imho the result is a good middle ground:

![Oct-16-2020 16-16-06](https://user-images.githubusercontent.com/2070010/96276657-3496ae00-0fcb-11eb-91e8-76c1e12acd52.gif)

(animation slowed to 10%)

```css
.edit-site-header {
	padding-left: 60px;
	transition: padding-left 100ms linear;
}

body.is-navigation-sidebar-open {
	.edit-site-header {
		padding-left: 0;
		transition: padding-left 20ms linear;
	}
}
```

I guess we might poke around with the transitions times to find an even better spot, especially for the slide out, but still... What do you think?

---

Also I wonder if we these header transitions should be moved to the sidebar's styles instead. 🤔

Reply:
This was my very first idea. I ditched it because it seemed to be impossible to get it perfect 😄 And another reason is, we can't really use anything other than `linear` transitions. Anyway, if we stick to linear then it's perfect:

**slide-in**

The slide in transition can be calculated easily. Since our toggle is `60px` and our sidebar is `300px`. The sidebar takes this time to reach the toggle's width: `toggleWidth / sidebarWidth * transitionTime`, in our case `60px / 300px * 100ms` which is `20ms`. We want to shrink `padding-left` until the sidebar reached the toggle's width. We start the transition at the same time as we start to transition the sidebar (since the toggle is on the left side), for the time it takes to reach the toggle's width: 
```css
transition: padding-left 20ms linear;
transition-delay: 0ms;
```

**slide-out**

Using the math above, we can calculate when we should start transitioning the `padding-left` and for how long. Since our sidebar takes `20ms` to reach toggle's width, we need to delay the transition by `transitionTime - timeToToggleWidth`, in our case `100ms - 20ms`. It takes `80ms` to reach the toggle's width from a fully opened sidebar. If we delay our transition by `80ms`, that means we aren't shrinking the `padding-left` until we reached the toggle's width. Once we reached, we need to run the transition for `20ms`, which is how long it takes to go from toggle's width to zero: 
```css
transition: padding-left 20ms linear;
transition-delay: 80ms;
```

**Result:**

![ezgif com-gif-maker (4)](https://user-images.githubusercontent.com/2256104/96417784-152d9a00-11f2-11eb-86fc-82a114f2ba50.gif)